### PR TITLE
Remove woocommerce-legacy.scss

### PR DESCRIPTION
**Pull request by Aljullu**
[Storefront 2.5.7 supports WC >=4.0](https://docs.woocommerce.com/document/storefront-requirements-version-support/), so there is no reason to keep stylesheets for legacy versions of WC.

### How to test the changes in this Pull Request:
Verify there are no regressions in styles.

### Changelog

> Fix – Removed `woocommerce-legacy.css` stylesheets for unsupported versions of WooCommerce (< 4.0). #1368

### CODE REVIEW

**Comment by haszari** [comment]
Note: I'm testing this with WooCommerce Blocks deactivated because of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2646. In any case we'll need to give this a thorough test before release.

File: inc/woocommerce/class-storefront-woocommerce.php [hunk 1]
```diff
@@ -150,11 +150,6 @@ public function woocommerce_scripts() {
 			if ( ! class_exists( 'Storefront_Sticky_Add_to_Cart' ) && is_product() ) {
 				wp_register_script( 'storefront-sticky-add-to-cart', get_template_directory_uri() . '/assets/js/sticky-add-to-cart' . $suffix . '.js', array(), $storefront_version, true );
 			}
-
-			if ( defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '3.3', '<' ) ) {
```
Comment (haszari) [inline inc/woocommerce/class-storefront-woocommerce.php:154]: We should probably search the code for other uses of `WC_VERSION` - maybe there's other stuff we can tidy up.

Comment (Aljullu) [reply inc/woocommerce/class-storefront-woocommerce.php:154]: Sure thing! I found this one while working on #1369, but there might be others.

Comment (haszari) [reply inc/woocommerce/class-storefront-woocommerce.php:154]: #1373 💥

**Pull request merged**